modin/numpy/arr.py

```diff
@@ -0,0 +1,333 @@
+                "subok": subok,
+                "ndmin": ndmin,
+                "like": like,
+            }
+            for key, value in rcvd_kwargs.copy().items():
+                if value == expected_kwargs[key]:
+                    rcvd_kwargs.pop(key)
+            arr = numpy.array(object, **rcvd_kwargs)
+            self._ndim = len(arr.shape)
+            if self._ndim > 2:
+                ErrorMessage.not_implemented(
+                    "NumPy arrays with dimensions higher than 2 are not yet supported."
+                )
+            import modin.pandas as pd
+
+            self._query_compiler = pd.DataFrame(arr)._query_compiler
+
+    def _absolute(
+        self,
+        out=None,
+        where=True,
+        casting="same_kind",
+        order="K",
+        dtype=None,
+        subok=True,
+    ):
+        result = self._query_compiler.abs()
+        return array(_query_compiler=result, _ndim=self._ndim)
+
+    __abs__ = _absolute
+
+    def _binary_op(self, other):
+        broadcast = self._ndim != other._ndim
+        if broadcast:
+            # In this case, we have a 1D object doing a binary op with a 2D object
+            caller = self if self._ndim == 2 else other
+            callee = other if self._ndim == 2 else self
+            return (caller, callee, caller._ndim, {"broadcast": broadcast, "axis": 1})
+        else:
+            if self.shape != other.shape:
+                # In this case, we either have two mismatched objects trying to do an operation
+                # or a nested 1D object that must be broadcasted trying to do an operation.
+                matched_dimension = None
+                if self.shape[0] == other.shape[0]:
+                    matched_dimension = 0
+                elif self.shape[1] == other.shape[1]:
+                    matched_dimension = 1
+                if matched_dimension is not None:
+                    if (
+                        self.shape[matched_dimension ^ 1] == 1
+                        or other.shape[matched_dimension ^ 1] == 1
+                    ):
+                        # caller = self if other.shape[matched_dimension ^ 1] == 1 else other
+                        # callee = other if other.shape[matched_dimension ^ 1] == 1 else self
+                        return (self, other, self._ndim, {"broadcast": True, "axis": 1})
+                else:
+                    raise ValueError(
+                        f"operands could not be broadcast together with shapes {self.shape} {other.shape}"
+                    )
+            else:
+                return (self, other, self._ndim, {"broadcast": False})
+
+    def _add(
+        self,
+        x2,
+        out=None,
+        where=True,
+        casting="same_kind",
+        order="K",
+        dtype=None,
+        subok=True,
+    ):
+        caller, callee, new_ndim, kwargs = self._binary_op(x2)
+        result = caller._query_compiler.add(callee._query_compiler, **kwargs)
```

Comment:
what is `kwargs` and why no real args like `where` are passed through? 😉 

same in lots of other places

Reply:
We're ignoring the more complicated kwargs from numpy for now, but the kwargs that binary_ops gives us helps the query compiler understand how to broadcast.

Reply:
The thing is, `kwargs` is not defined anywhere, and real `_add` args are simply ignored. Feels wrong.

Reply:
This is in the old code - the new code handles the actual __add__ kwargs.